derive/src/utils.rs

```diff
@@ -465,6 +479,21 @@ fn check_variant_attribute(attr: &Attribute) -> syn::Result<()> {
 	}
 }
 
+// Ensure a variant discriminant, if provided, can be parsed into
+// something in the range 0..255.
+fn check_variant_discriminant(discriminant: &Expr) -> syn::Result<()> {
+	if let Expr::Lit(ExprLit { lit: Lit::Int(lit_int), .. }) = discriminant {
+		lit_int.base10_parse::<u8>().map(|_| ()).map_err(|_| {
+			syn::Error::new(lit_int.span(), "Discriminant index must be in the range 0..255")
+		})
+	} else {
+		Err(syn::Error::new(
+			discriminant.span(),
+			"Discriminant must be an integer literal in the range 0..255",
```

Comment:
Forcing this to be an integer is a breaking change.

Potential fix: https://github.com/paritytech/parity-scale-codec/pull/695